targets/XPlatCppSdk/source/code/include/playfab/PlayFabWinHttpPlugin.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <playfab/PlayFabCallRequestContainer.h>
+#include <playfab/PlayFabPluginManager.h>
+#include <thread>
+#include <mutex>
+#include <windows.h>
+#include <winhttp.h>
+
+namespace PlayFab
+{
+    /// <summary>
+    /// PlayFabWinHttpPlugin is an https implementation to interact with PlayFab services using WinHTTP.
+    /// </summary>
+    class PlayFabWinHttpPlugin : public IPlayFabHttpPlugin
+    {
+    public:
+        PlayFabWinHttpPlugin();
+        PlayFabWinHttpPlugin(const PlayFabWinHttpPlugin& other) = delete;
+        PlayFabWinHttpPlugin(PlayFabWinHttpPlugin&& other) = delete;
+        PlayFabWinHttpPlugin& operator=(PlayFabWinHttpPlugin&& other) = delete;
+        virtual ~PlayFabWinHttpPlugin();
+
+        virtual void MakePostRequest(std::unique_ptr<CallRequestContainerBase> requestContainer) override;
+        virtual size_t Update() override;
+
+    protected:
+        virtual void ExecuteRequest(std::unique_ptr<CallRequestContainer> requestContainer);
+        virtual std::string GetUrl(CallRequestContainer& requestContainer) const;
+        virtual void SetPredefinedHeaders(CallRequestContainer& requestContainer, HINTERNET hRequest);
+        virtual bool GetBinaryPayload(CallRequestContainer& requestContainer, LPVOID& payload, DWORD& payloadSize) const;
+        virtual void ProcessResponse(CallRequestContainer& requestContainer, const int httpCode);
+        void WorkerThread();
+        void HandleCallback(std::unique_ptr<CallRequestContainer> requestContainer);
+        void HandleResults(std::unique_ptr<CallRequestContainer> requestContainer);
+        void SetErrorInfo(CallRequestContainer& requestContainer, const std::string& errorMessage, const int httpCode = 0) const;
+
+        std::thread workerThread;
+        std::mutex httpRequestMutex;
+        bool threadRunning;
```

Comment:
>  bool threadRunning; [](start = 6, length = 21)

should this be an atomic bool ?

Reply:
in most cases making bool atomic is an overkill. With this particular bool i am not introducing any new pattern here. If you insist on making this atomic we should change that in all 6 other plugins too.

---
In reply to: [244876563](https://github.com/PlayFab/SDKGenerator/pull/368#discussion_r244876563) [](ancestors = 244876563)

Reply:
Yes - let's change them all through

---
In reply to: [244877114](https://github.com/PlayFab/SDKGenerator/pull/368#discussion_r244877114) [](ancestors = 244877114,244876563)